model-optimizer/extensions/front/kaldi/tdnn_component_replacer.py

```diff
@@ -0,0 +1,108 @@
+"""
+ Copyright (C) 2018-2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from extensions.ops.MatMul import FullyConnected
+from mo.front.common.replacement import FrontReplacementPattern
+from mo.graph.graph import Graph, Node
+from mo.graph.graph import rename_nodes
+from mo.ops.concat import Concat
+from mo.ops.const import Const
+from mo.ops.memoryoffset import MemoryOffset
+
+
+class TdnnComponentReplacer(FrontReplacementPattern):
+    '''
+    'unfuse' TdnnComponent into MemoryOffsets, Concats and FullyConected layer
+
+    BEFORE:                 john-doe
+                               |
+                          TDNN-Component
+                               |
+    _________________________________________________________________________
+
+    After:                  john-doe
+            ___________________|___________________________
+           /                   |                           \
+    memoryOffset(t1)     MemoryOffset(t2)    ...    MemoryOffset(tk)
+           \_____________ _____|__________________________/
+                            Concat
+                               |
+                          FullyConnected
+                               |
+    '''
+    enabled = True
+    run_not_recursively = True
+
+    def run_after(self):
+        from extensions.front.restore_ports import RestorePorts
+        return [RestorePorts]
+
+    def run_before(self):
+        from extensions.front.kaldi.split_memoryoffsets import SplitMemoryOffsets
+        from extensions.front.kaldi.memory_offset_adjustment import MemoryOffsetAdjustment
+        return [SplitMemoryOffsets, MemoryOffsetAdjustment]
+
+    def pattern(self):
+        return dict(
+            nodes=[
+                ('tdnncomponent', dict(kind='op', op='tdnncomponent'))
+            ],
+            edges=[]
+        )
+
+    @staticmethod
+    def replace_pattern(graph: Graph, match: dict):
+        tdnn_node: Node = match['tdnncomponent']
```

Comment:
Do not use pattern. Instead use something like this: `for tdnn_node in graph.get_op_nodes(op='tdnncomponent'):`

Reply:
Done. Please review